src/Lykke.Job.NeoGasDistributor.Services/BalanceService.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Lykke.Job.NeoGasDistributor.Domain;
+using Lykke.Job.NeoGasDistributor.Domain.Repositories;
+using Lykke.Job.NeoGasDistributor.Domain.Services;
+
+namespace Lykke.Job.NeoGasDistributor.Services
+{
+    [UsedImplicitly]
+    public class BalanceService : IBalanceService
+    {
+        private readonly IBalanceUpdateRepository _balanceUpdateRepository;
+        private readonly ISnapshotRepository _snapshotRepository;
+
+        
+        public BalanceService(
+            IBalanceUpdateRepository balanceUpdateRepository,
+            ISnapshotRepository snapshotRepository)
+        {
+            _balanceUpdateRepository = balanceUpdateRepository;
+            _snapshotRepository = snapshotRepository;
+        }
+
+        
+        public async Task CreateSnapshotAsync(
+            DateTime from,
+            DateTime to)
+        {
+            var snapshotBalances = (await _balanceUpdateRepository
+                .GetAsync(from, to))
+                .OrderBy(x => x.EventTimestamp)
+                .GroupBy(x => x.WalletId)
+                .ToDictionary(x => x.Key, x => x.Select(y => y.NewBalance).Last());
```

Comment:
.ToDictionary(x => x.Key, x => x..OrderBy(p => p.EventTimestamp)Select(y => y.NewBalance).Last());

is more clear as for me